server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1094,6 +1105,11 @@ private static class IndexMetadataDiff implements Diff<IndexMetadata> {
             );
             rolloverInfos = DiffableUtils.readJdkMapDiff(in, DiffableUtils.getStringKeySerializer(), ROLLOVER_INFO_DIFF_VALUE_READER);
             isSystem = in.readBoolean();
+            if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+                context = in.readOptionalWriteable(Context::new);
```

Comment:
Although DiffableUtils is not used to compute the IndexMetadata upserts, index metadata version is used to compute the changed IndexMetadata list. We do not directly accessed the fields in IndexMetadata.  
Also, the IndexMetadata serialization is done using the same toXContent method which is already modified here. So we should be good.
@shwetathareja But I am thinking whether we should increment the remote state codec version.

Reply:
@soosinha yes we may have to bump it up

Reply:
@soosinha @shwetathareja Since we are recommending a version bump for remote state codec, I'm assuming there will be some failures observed once we start writing the new metadata. To verify, I can run tests on top of the PR https://github.com/opensearch-project/OpenSearch/pull/15478 to ensure tests run with settings enabled (copying @shiv0408 as he provided inputs on how to run the tests)

Reply:
@soosinha Do you mean bumping the `INDEX_METADATA_CURRENT_CODEC_VERSION` right? I don't see any point of bumping the codec version of Cluster Metadata Manifest as there wont be any change in manifest serde. Also if required, we will need to add handling of this changed codec version of IndexMetadata.

Reply:
I agree if we dont need any ser/de change for IndexMetadata Handling for remote cluster state we dont need bump up the codec version. It would be good to test with context end-to-end.

Reply:
Resolving it for now.